### PR TITLE
test: improve e2e test troubleshooting

### PR DESCRIPTION
**Pull request by simonpasquier**
## Description

This change gathers and dumps useful information when an end-to-end test fails.

For now the information is dumped to the test output, the small downside is that it clutters a bit the job's logs in case a test fails. A future improvement could be to save the data to a directory and upload it to GitHub artifacts.

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [X] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -366,31 +366,31 @@ test-e2e: test/instrumented-sample-app/certs/cert.pem test/instrumented-sample-a
 
 .PHONY: test-e2e-alertmanager
 test-e2e-alertmanager:
-	EXCLUDE_PROMETHEUS_TESTS=exclude EXCLUDE_PROMETHEUS_ALL_NS_TESTS=exclude EXCLUDE_THANOSRULER_TESTS=exclude EXCLUDE_OPERATOR_UPGRADE_TESTS=exclude EXCLUDE_FEATURE_GATED_TESTS=exclude EXCLUDE_PROMETHEUS_UPGRADE_TESTS=exclude $(MAKE) test-e2e
+	EXCLUDE_ALERTMANAGER_TESTS= EXCLUDE_PROMETHEUS_TESTS=exclude EXCLUDE_PROMETHEUS_ALL_NS_TESTS=exclude EXCLUDE_THANOSRULER_TESTS=exclude EXCLUDE_OPERATOR_UPGRADE_TESTS=exclude EXCLUDE_FEATURE_GATED_TESTS=exclude EXCLUDE_PROMETHEUS_UPGRADE_TESTS=exclude $(MAKE) test-e2e
```
Comment (slashpai) [inline Makefile:369]: Did you leave `EXCLUDE_ALERTMANAGER_TESTS= ` to test if logging working?

File: test/framework/prometheus.go [hunk 2]
```diff
@@ -436,7 +436,7 @@ func (f *Framework) CreatePrometheusAndWaitUntilReady(ctx context.Context, ns st
 		return nil, fmt.Errorf("creating %d Prometheus instances failed (%v): %v", ptr.Deref(p.Spec.Replicas, 1), p.Name, err)
 	}
 
-	result, err = f.WaitForPrometheusReady(ctx, result, 5*time.Minute)
+	result, err = f.WaitForPrometheusReady(ctx, result, 1*time.Minute)
```
Comment (slashpai) [inline test/framework/prometheus.go:439]: Guess it was changed for testing?

**Review by slashpai — COMMENTED** [review]
Looks good to me, I checked one of the action failed it shows clear message

```
=== NAME  TestAllNS
    errgroup.go:78: Test TestAllNS
    context.go:80: timestamp='2024-09-23 15:16:58 +0000 UTC' namespace="allns-sk9ufw-1-6a1f7a28" reason="Scheduled" message="Successfully assigned allns-sk9ufw-1-6a1f7a28/prometheus-operator-54bdfcd48c-9pt5k to e2e-worker2"
        timestamp='2024-09-23 15:16:58 +0000 UTC' namespace="allns-sk9ufw-1-6a1f7a28" reason="Pulled" message="Container image \"quay.io/prometheus-operator/prometheus-operator:9770a11\" already present on machine"
        timestamp='2024-09-23 15:16:58 +0000 UTC' namespace="allns-sk9ufw-1-6a1f7a28" reason="Created" message="Created container prometheus-operator"
        timestamp='2024-09-23 15:16:58 +0000 UTC' namespace="allns-sk9ufw-1-6a1f7a28" reason="Started" message="Started container prometheus-operator"
        timestamp='2024-09-23 15:16:58 +0000 UTC' namespace="allns-sk9ufw-1-6a1f7a28" reason="SuccessfulCreate" message="Created pod: prometheus-operator-54bdfcd48c-9pt5k"
        timestamp='2024-09-23 15:17:01 +0000 UTC' namespace="allns-sk9ufw-1-6a1f7a28" reason="Scheduled" message="Successfully assigned allns-sk9ufw-1-6a1f7a28/prometheus-operator-admission-webhook-549c6df475-7pzfr to e2e-worker2"
        timestamp='2024-09-23 15:17:02 +0000 UTC' namespace="allns-sk9ufw-1-6a1f7a28" reason="Pulled" message="Container image \"quay.io/prometheus-operator/admission-webhook:9770a11\" already present on machine"
        timestamp='2024-09-23 15:17:02 +0000 UTC' namespace="allns-sk9ufw-1-6a1f7a28" reason="Created" message="Created container prometheus-operator-admission-webhook"
        timestamp='2024-09-23 15:17:02 +0000 UTC' namespace="allns-sk9ufw-1-6a1f7a28" reason="Started" message="Started container prometheus-operator-admission-webhook"
        timestamp='2024-09-23 15:17:01 +0000 UTC' namespace="allns-sk9ufw-1-6a1f7a28" reason="Scheduled" message="Successfully assigned allns-sk9ufw-1-6a1f7a28/prometheus-operator-admission-webhook-549c6df475-82wb9 to e2e-worker"
        timestamp='2024-09-23 15:17:02 +0000 UTC' namespace="allns-sk9ufw-1-6a1f7a28" reason="Pulled" message="Container image \"quay.io/prometheus-operator/admission-webhook:9770a11\" already present on machine"
        timestamp='2024-09-23 15:17:02 +0000 UTC' namespace="allns-sk9ufw-1-6a1f7a28" reason="Created" message="Created container prometheus-operator-admission-webhook"
        timestamp='2024-09-23 15:17:02 +0000 UTC' namespace="allns-sk9ufw-1-6a1f7a28" reason="Started" message="Started container prometheus-operator-admission-webhook"
        timestamp='2024-09-23 15:17:01 +0000 UTC' namespace="allns-sk9ufw-1-6a1f7a28" reason="SuccessfulCreate" message="Created pod: prometheus-operator-admission-webhook-549c6df475-82wb9"
        timestamp='2024-09-23 15:17:01 +0000 UTC' namespace="allns-sk9ufw-1-6a1f7a28" reason="SuccessfulCreate" message="Created pod: prometheus-operator-admission-webhook-549c6df475-7pzfr"
        timestamp='2024-09-23 15:17:01 +0000 UTC' namespace="allns-sk9ufw-1-6a1f7a28" reason="ScalingReplicaSet" message="Scaled up replica set prometheus-operator-admission-webhook-549c6df475 to 2"
        timestamp='2024-09-23 15:16:58 +0000 UTC' namespace="allns-sk9ufw-1-6a1f7a28" reason="ScalingReplicaSet" message="Scaled up replica set prometheus-operator-54bdfcd48c to 1"
```

File: Makefile (hunk 1 above)
Comment (slashpai) [reply Makefile:369]: similarly for other test as well

File: test/framework/prometheus.go (hunk 2 above)
Comment (simonpasquier) [reply test/framework/prometheus.go:439]: Yes I'll need to revert :)

File: Makefile (hunk 1 above)
Comment (simonpasquier) [reply Makefile:369]: IMHO it's more explicit and easier to understand which sub test suite is executed.

**Review by slashpai — APPROVED** [review]

**Pull request merged**